### PR TITLE
Petset e2es

### PR DESCRIPTION
**Pull request by bprashanth**
Basic petset cluster deploy/restart e2e, basically does:

``` bash
for pet in zookeeper msyql redis; do 
  write a key to some node
  restart cluster
  read key from different node
done
```

@jlowdermilk @smarterclayton whichever one of you has time (or whoever merge robot picks)

Images need to be replaced once: https://github.com/kubernetes/contrib/pull/921 goes in

### CODE REVIEW

File: test/e2e/petset.go [hunk 1]
```diff
@@ -32,114 +35,348 @@ import (
...
 			Expect(err).NotTo(HaveOccurred())
-		}()
+		})
 
-		petMounts := []api.VolumeMount{{Name: "datadir", MountPath: "/data/"}}
-		podMounts := []api.VolumeMount{{Name: "home", MountPath: "/home"}}
-		ps := newPetSet(psName, ns, headlessSvcName, 3, petMounts, podMounts, labels)
-		_, err := c.Apps().PetSets(ns).Create(ps)
-		Expect(err).NotTo(HaveOccurred())
+		It("provide basic identity [Feature:PetSet]", func() {
```
Comment (jlowdermilk) [inline test/e2e/petset.go]: nit: "should provide ..."

File: test/e2e/petset.go [hunk 2]
```diff
@@ -32,114 +35,348 @@ import (
...
-		headlessService := createServiceSpec(headlessSvcName, true, labels)
-		_, err = c.Services(ns).Create(headlessService)
-		Expect(err).NotTo(HaveOccurred())
 	})
 
-	It("provide basic identity [Feature:PetSet]", func() {
-		By("creating petset " + psName + " in namespace " + ns)
-		defer func() {
-			err := c.Apps().PetSets(ns).Delete(psName, nil)
+	framework.KubeDescribe("Basic PetSet functionality", func() {
```
Comment (jlowdermilk) [inline test/e2e/petset.go]: You'll want to add `framework.SkipUnlessProviderIs("gce")` here.

Comment (bprashanth) [inline test/e2e/petset.go]: added to beforeEach

**Comment by jlowdermilk** [comment]
LGTM. Squash, and I'll add label.

**Comment by bprashanth** [comment]
Squashed. Sadly, because of conflicting kube-dns changes these e2es won't work *right now* (they're not enabled yet though). I also still need the images in contrib, so I assigned that pr to you (https://github.com/kubernetes/contrib/pull/921).

**Comment by bprashanth** [comment]
Inheriting LGTM, only change was a replacement of the `erkules/galera:basic` with `gcr.io/google_containers/mysql-galera:e2e`.

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this issue: #IGNORE (tests passed, upload results failed).

**Comment by bprashanth** [comment]
Just a govet fix and rebase, re-applying lgtm

**Comment by a-robinson** [comment]
Should this be part of the 1.3 milestone, or no?

**Comment by bprashanth** [comment]
Yes, added

**Pull request merged**